prov/rxd/src/rxd_atomic.c

```diff
@@ -235,7 +235,7 @@ static ssize_t rxd_atomic_inject(struct fid_ep *ep_fid, const void *buf,
 	if (ofi_cirque_isfull(rxd_ep->util_ep.tx_cq->cirq))
 		goto out;
 
-	rxd_addr = rxd_ep_av(rxd_ep)->fi_addr_table[addr];
+	rxd_addr = rxd_ep_av(rxd_ep)->fi_addr_table[dest_addr];
```

Comment:
It looks like it's wrong in rxd_generic_atomic (line 137). Could you fix it there, too, please? Thanks!

Reply:
I dont see a dest_addr argument  in that function. there is only a addr of type fi_addr_t. is that incorrect?

Reply:
Oh yup, yup. You're right. Nevermind!